### ISSUE TITLE
SMG issues: Fire-rate randomly increases, ammunition refunded partially when fire-rate increases

### ISSUE THREAD

**Issue by DetectDefective** [yvt/openspades#560]
I remember spotting this a while back, but I only posted about it on the BnS forums.

The SMG has some weird behaviour, where it will start firing faster than it normally can, and after reloading from empty, the spare ammo will actually have a few bullets refunded.

I have uploaded a video of the bug in action, [here](https://www.youtube.com/watch?v=dii5Qynt5Es). Annotations are in place to mark/point out specifics.

**Comment by damienflament** [comment]
@DetectDefective Which OpenSpades version are you using ? I'm using the latest stable version without mods but french translations.
And I'm not able to reproduce your issue.

Could this be server configuration related ? I know blocks placement speed can be customized on server side.

**Comment by DetectDefective** [comment]
I'm using 0.1.0, the version that was released about a month ago. Before this, when I first noticed this issue, I was using 0.0.10, which was from January 2014.

I'm bringing up the old version because I actually forgot that I had uploaded a [video](https://www.youtube.com/watch?v=RLNlb_y8nPw) about this bug, way back when I made that forum post.

As for server configurations, I'm not really certain. For the recent video, I'd been playing on aloha.pk's classicgen server. For the video in 2014, it was some random hallway CTF server.
